src/StoreMeta.h

```diff
@@ -9,63 +9,51 @@
 #ifndef SQUID_TYPELENGTHVALUE_H
 #define SQUID_TYPELENGTHVALUE_H
 
-class StoreEntry;
+#include "defines.h"
+#include "store/forward.h"
 
-// WTF?
-typedef class StoreMeta tlv;
+#include <iosfwd>
+#include <limits>
 
 /**
- \ingroup SwapStoreAPI
- * XXX: for critical lists like this we should use A=64,B=65 etc to enforce and reserve values.
- \note NOTE!  We must preserve the order of this list!
  *
  \section StoreSwapMeta Store "swap meta" Description
  \par
- * "swap meta" refers to a section of meta data stored at the beginning
- * of an object that is stored on disk.  This meta data includes information
- * such as the object's cache key (MD5), URL, and part of the StoreEntry
- * structure.
+ * Store "swap meta" is Store entry metadata stored at the beginning of each
+ * cache_dir entry. This entry-dependent metadata typically includes entry cache
+ * key, URL, size, and various timestamps. Copies of this information may also
+ * be present in Squid process memory, readily accessible via StoreEntry and
+ * MemObject pointers, but swap metadata is about cached entry information
+ * stored in serialized form meant to cross process and instance boundaries.
```

Comment:
> Amos at https://github.com/squid-cache/squid/pull/1067#pullrequestreview-996206115: The term "Swap" specifically restricts scope to on-disk objects, whereas these metadata are about objects that never go near a disk. Please return to "Store Metadata" as more accurate terminology self-documenting what these TLV contain.

Addressed in ec39176 by polishing documentation explaining that this Store swap metadata is indeed restricted to serialized "on-disk" bytes. Store metadata is a more general concept (that includes swap metadata, of course, but most Store entries are never swapped in or out and, hence, do not touch this serialization-related code). We should not change this. The Store-vs-swap difference is meaningful and useful here. We are working specifically with swap metadata (fields) here. Some of them are, of course, copied into in-memory StoreEntry and MemObject fields, where their values may even go out of sync with swap metadata.